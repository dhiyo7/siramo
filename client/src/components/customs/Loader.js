import React, { Component } from 'react'
import {
  StyleSheet,
  View,
  Modal,
  ActivityIndicator
} from 'react-native'
import * as Progress from 'react-native-progress'

const Loader = props => {
  const {
    loading,
    ...attributes
  } = props

  return (
    <Modal 
      visible={loading}
      animationType={'none'}
      transparent={true}
      onRequestClose={() => {console.log('close modal')}}
    >
      <View style={styles.modalBackGround}>
        <View style={styles.ActivityIndicatorWrapper}>
          {/* <ActivityIndicator
          
            animating={loading}
          /> */}
          <Progress.CircleSnail color={['#90CAF9', '#FFAB91', '#A5D6A7']} size={100} thickness={4} spinDuration={10000}/>
        </View>
      </View>
    </Modal>
  )
}

const styles = StyleSheet.create({
  modalBackGround: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    backgroundColor: '#00000040'
  },
  ActivityIndicatorWrapper: {
    borderRadius: 10,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around'
  }
})

export default Loader