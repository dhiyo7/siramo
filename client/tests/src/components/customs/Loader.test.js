import React from 'react'
import {
  View, Text, TouchableOpacity, 
  TextInput, StyleSheet, Picker
} from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import * as Progress from 'react-native-progress'
import Loader from '../../../../src/components/customs/Loader'
import { wrap } from 'module'

Enzyme.configure({ adapter: new Adapter()})

let wrapper

describe('<Loader /> snapshot Testing', () => {
  it('snapshot works!', () => {
    const tree = renderer.create(<Loader loading={true} />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<Loader />', () => {
  wrapper = shallow(<Loader loading={true} />)
  it('should render <FarmDetail />', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<Loader /> testing props', () => {
  it('visible value should be true', () => {
    wrapper = shallow(<Loader loading={true} />)
    // console.log(wrapper.get(0).props.children.props.children.props)
    expect(wrapper.get(0).props.visible).toEqual(true)
  })
  it('visible value should be false', () => {
    wrapper = shallow(<Loader loading={false} />)
    expect(wrapper.get(0).props.visible).toEqual(false)
  })
})

describe('<Loader /> Child Testing', () => {
  it('<Loader /> Modal onRequestClose', () => {
    wrapper = shallow(<Loader loading={true} />)
    let reqClose = wrapper.get(0).props.onRequestClose() 
    expect(reqClose).toEqual('close modal')
  })
})