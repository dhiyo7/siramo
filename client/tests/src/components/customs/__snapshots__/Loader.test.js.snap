// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> snapshot Testing snapshot works! 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "display": "flex",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        collapsable={undefined}
        spinDuration={10000}
        style={
          Object {
            "backgroundColor": "transparent",
            "overflow": "hidden",
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <ARTSurfaceView
          style={
            Array [
              undefined,
              Object {
                "height": 100,
                "width": 100,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                46.206330167420525,
                12.189841719435023,
                3,
                47.46693141297555,
                12.063359706831747,
                48.73306937845552,
                12,
                49.99999999999999,
                12,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.5647058823529412,
                0.792156862745098,
                0.9764705882352941,
                1,
              ]
            }
            strokeCap={1}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={8}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
      </View>
    </View>
  </View>
</Modal>
`;
